### DISCUSSION TITLE
How can I add the possibility to serialize input/output data with Protobuf,Messagepack,Avro etc.?

### DISCUSSION THREAD

**Discussion by junoriosity** [fastapi/fastapi#10533] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Description

I would like to shorten both the input and the output for increasing speed and lowering the bandwidth.

The most natural candidate would be to serialize the input and output JSON via Protobof, Messagepack, Avro or something else.

However, I did not find anything that matches my needs.

Do you know how to get such a thing done?

### Operating System

macOS

### FastAPI Version

0.104.0

### Pydantic Version

2.4.0

### Python Version

3.12

**Comment by jgould22** [comment]
You would create a custom Request and APIRoute class as described here https://fastapi.tiangolo.com/how-to/custom-request-and-route/#custom-request-and-apiroute-class.
